Flaky test: test_reusable_strategies_are_all_reusable

Ran into this today while testing some unrelated changes:

```
$ pytest tests/cover/test_reusable_values.py --hypothesis-seed=254419159797874700780956937497361070957
============================= test session starts ==============================
platform darwin -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/var/folders/54/vs3dcnhx7rlddhj58w5mzlz40000gn/T/tmp6crzei5y/examples')
rootdir: /Users/stuart/Dev/hypothesis, inifile: pytest.ini
plugins: xdist-1.27.0, forked-1.0.2, flaky-3.5.3, hypothesis-4.14.0
collected 6 items                                                              

tests/cover/test_reusable_values.py F.....                               [100%]

=================================== FAILURES ===================================
__________________ test_reusable_strategies_are_all_reusable ___________________
Traceback (most recent call last):
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/tests/cover/test_reusable_values.py", line 64, in test_reusable_strategies_are_all_reusable
    @given(reusable)
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/core.py", line 1017, in wrapped_test
    raise the_error_hypothesis_found
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/core.py", line 980, in wrapped_test
    state.run()
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/core.py", line 653, in run
    runner.run()
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/internal/conjecture/engine.py", line 387, in run
    self._run()
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/internal/conjecture/engine.py", line 720, in _run
    self.generate_new_examples()
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/internal/conjecture/engine.py", line 704, in generate_new_examples
    self.test_function(data)
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/internal/conjecture/engine.py", line 140, in test_function
    self.__stoppable_test_function(data)
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/internal/conjecture/engine.py", line 123, in __stoppable_test_function
    self._test_function(data)
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/core.py", line 599, in evaluate_test_data
    result = self.execute(data)
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/core.py", line 568, in execute
    result = self.test_runner(data, run)
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/executors.py", line 56, in default_new_style_executor
    return function(data)
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/core.py", line 566, in run
    return test(*args, **kwargs)
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/tests/cover/test_reusable_values.py", line 64, in test_reusable_strategies_are_all_reusable
    @given(reusable)
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/core.py", line 529, in test
    result = self.test(*args, **kwargs)
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/tests/cover/test_reusable_values.py", line 67, in test_reusable_strategies_are_all_reusable
    s.validate()
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/searchstrategy/strategies.py", line 377, in validate
    self.do_validate()
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/searchstrategy/lazy.py", line 122, in do_validate
    w = self.wrapped_strategy
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/searchstrategy/lazy.py", line 112, in wrapped_strategy
    base = self.__function(*self.__args, **self.__kwargs)
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/_strategies.py", line 560, in floats
    note_deprecation(msg, since="2019-03-19")
  File "/Users/stuart/Dev/hypothesis/hypothesis-python/src/hypothesis/_settings.py", line 785, in note_deprecation
    warnings.warn(warning, stacklevel=2)
hypothesis.errors.HypothesisDeprecationWarning: There are no 64-bit floating-point values between min_value=0.0 and max_value=-0.0
===Flaky Test Report===


===End Flaky Test Report===
========================== slowest 20 test durations ===========================
0.55s call     hypothesis-python/tests/cover/test_reusable_values.py::test_reusable_strategies_are_all_reusable
0.02s setup    hypothesis-python/tests/cover/test_reusable_values.py::test_recursion_does_not_break_reusability
0.02s setup    hypothesis-python/tests/cover/test_reusable_values.py::test_mutable_collections_do_not_have_reusable_values[strat0]
0.02s setup    hypothesis-python/tests/cover/test_reusable_values.py::test_mutable_collections_do_not_have_reusable_values[strat2]
0.02s setup    hypothesis-python/tests/cover/test_reusable_values.py::test_composing_breaks_reusability
0.02s setup    hypothesis-python/tests/cover/test_reusable_values.py::test_reusable_strategies_are_all_reusable
0.02s setup    hypothesis-python/tests/cover/test_reusable_values.py::test_mutable_collections_do_not_have_reusable_values[strat1]

(0.00 durations hidden.  Use -vv to show these durations.)
====================== 1 failed, 5 passed in 0.71 seconds ======================
```

--- a/hypothesis-python/tests/cover/test_reusable_values.py
+++ b/hypothesis-python/tests/cover/test_reusable_values.py
@@ -21,7 +21,7 @@ import pytest
 
 import hypothesis.strategies as st
 from hypothesis import example, given, reject
-from hypothesis.errors import InvalidArgument
+from hypothesis.errors import HypothesisDeprecationWarning, InvalidArgument
 
 base_reusable_strategies = (
     st.text(),
@@ -65,7 +65,7 @@ assert not reusable.is_empty
 def test_reusable_strategies_are_all_reusable(s):
     try:
         s.validate()
-    except InvalidArgument:
+    except (InvalidArgument, HypothesisDeprecationWarning):
         reject()
 
     assert s.has_reusable_values

